### PR TITLE
PHP 7.1+, PHPUnit 8 support

### PR DESCRIPTION
**Pull request by codemasher**
Hi, can we have PHPUnit 8 support please? 
image

### CODE REVIEW

File: test/RequestFactoryTestCase.php [hunk 1]
```diff
@@ -26,7 +26,7 @@ abstract class RequestFactoryTestCase extends TestCase
      */
     abstract protected function createUri($uri);
 
-    public function setUp()
+    public function setUp():void
```
Comment (shadowhand) [inline test/RequestFactoryTestCase.php]: There should be a space between the `:` and the return type.

**Review by shadowhand — COMMENTED** [review]
I'm not really sure why this is necessary. This package should only be used for testing http-factory implementations.

**Comment by codemasher** [comment]
It's necessary in case when a package uses PHPUnit 8 for unit testing and also wants to use your Integration tests. The return types for `TestCase::setUp():void` etc. are mandatory now.

**Review by shadowhand — APPROVED** [review]

**Comment by shadowhand** [comment]
@http-interop/http-factory-contributors anyone else want to look this over?

**Comment by Zegnat** [comment]
Is there a reason for dropping support for PHPUnit 6.5, other than the fact that it has reached EOL? I guess that is a fine reason, just wondering.

The other thing that puzzles me is why this drops support for PHPUnit < 7.5. If the suite is currently compatible with everything 7.0+, and the only new requirement is for PHP 7.1 because of the `void` return type, why require `^7.5` rather than sticking to the previously used `^7.0`?

For backwards compatibility with other projects that may want to run these tests, why not simply require PHPUnit `^6.5 || ^7.0 || ^8.0` and stay compatible with ourselves?

**Comment by codemasher** [comment]
PHPUnit 8 requires PHP 7.2, i left in PHPUnit 7.5 because it's the most recent supporting PHP 7.1. My reasoning was that this change is breaking anyway for older PHP versions and therefore would require a major version step - so why not throw out the EOL PHPunit as well?
But sure, i don't have a problem leaving the older PHPUnit versions in.

**Comment by shadowhand** [comment]
Thanks!

**Comment by codemasher** [comment]
Thank you! :tada:

**Pull request merged**